### DISCUSSION TITLE
How to get specific error data

### DISCUSSION THREAD

**Discussion by MrOtek598** [pocketbase/pocketbase#2389] [Q&A] [answered]
Hi,

Thank you so much for Pocketbase, I love this app !

I'm looking to handle PB errors in my Sveltekit app in hope to know exactly what fields are requied or already exists. Is this possible today in PB ? Something like : 
`{
{ code: 'validation_required', message: 'Missing required value.', field : 'name' },
{ code: 'validation_not_unique', message: 'Value must be unique.', field : 'username' },
}`

I found this discussion https://github.com/pocketbase/pocketbase/discussions/196. Is there still a bug about this or this is not yet possible with the js-sdk ?
 
Thank you, your projects are really inspiring for me and motivate me to learn Go. 
I hope you will get where you want to be with your works, you deserve it.

**Comment by ganigeorgiev** [comment]
> I'm looking to handle PB errors in my Sveltekit app in hope to know exactly what fields are requied or already exists. Is this possible today in PB ? 

I think this is already supported. You can access individual field errors (if there any) by inspecting the error response `data` map, aka. `err.response.data` (the error field name is a key in the map). If you are using the JS SDK, all api errors are standardized and returned as `ClientResponseError` (if you haven't already, for more info you can check [Error handling](https://github.com/pocketbase/js-sdk#error-handling)).

**Reply by MrOtek598** [reply]
Thank you for your quick answer !
Yes I have looked in the doc, for which by the way I love the simplicity and the clarity too !

To be short and simple, it seems for me that when PB has multiple unsatified fields in a collection, it returns an empty `data` map.
When there is multiple fields to satisfy but only one is not, it returns a filled `data` map but with no information of the specific unsatified field.

For example, I have these three fields : `name` (unique), `owner` (unique), `status` (requied) in a `try` `catch` directive : `...collection('domains').create({name, owner, status})`
 

1. When the data already exists in my collection, if I come again with the same information : `...collection('domains').create({name : 'example.com', owner : 'user1', status : 'prod'})` 
I expect PB to say to me that there is already a domain named `example.com` and already a `owner` `'user1'` for one domain in that collection but I get an empty `data` map in `err.response.data`.


2. If I give to PB the same `name` but with a new `owner` `'user2'` in that collection : `...collection('domains').create({name : 'example.com', owner : 'user1', status : 'prod'})`
I get a filled `data` map in `err.response.data` with :
`...{ code: 'validation_not_unique', message: 'Value must be unique.' }`
In that response, I have no idea of what unique field is it about, `name` or `owner` ?

Is this a normal behavior for now in the js-sdk ?

Thank you !

**Reply by ganigeorgiev** [reply, marked as answer]
I don't understand the above examples. The `create` statements are not valid since you are missing the field keys.

> I get a filled data map in err.response.data with :

The exact error field is a key in the map, aka. `err.response.data.yourField`.

I've tested it locally with PocketBase v0.15.2 and JS SDK v0.14.3 and it is working fine. Here is an example:

```js
try {
    await pb.collection("example").create({
       "title":       "ab", // require at least 3 characters
       "description": "",   // must be nonemtpy
    });
} catch (err) {
    console.log(err.response);
    // Output:
    // {
    //     "code": 400,
    //     "message": "Failed to create record.",
    //     "data": {
    //         "description": {
    //             "code": "validation_required",
    //             "message": "Missing required value."
    //         },
    //         "title": {
    //             "code": "validation_min_text_constraint",
    //             "message": "Must be at least 3 character(s)."
    //         }
    //     }
    // }
}
```

Note that authentication errors are intentionally generic to prevent to some extend users enumeration.
If the above doesn't help, please provide the request response that is being returned (_you can find it in the browser's Inspect tool > Network tab > Response_).

**Reply by MrOtek598** [reply]
I knew it was me ! I tried to take time to understand and now I see and it has always works... Thanks to you ! Sorry, for some reason, I have never take attention to the `data` map, especially this part `err.response.data.**yourfield** `

Thank you so much ! I wish you the best !

```javascript
  const name = 'example.com' // unique
  const owner = 'user1' // unique
  const status = 'prod' // requied

  try {
    const record = await locals.pb.collection('domains').create({name, owner, status})
  } catch (error) {
    console.log(error.response.data)
    // Output :
    // {
    //   name: { code: 'validation_not_unique', message: 'Value must be unique.' },
    //   owner: { code: 'validation_not_unique', message: 'Value must be unique.' }
    // }
  }
```
